src/io/eval.rs

```diff
@@ -42,12 +42,16 @@ where
         let reveal_plaintexts = [vec![true; packed_input_size - 1], vec![true; 1]].concat();
         #[cfg(not(feature = "test"))]
         let reveal_plaintexts = [vec![true; packed_input_size - 1], vec![false; 1]].concat();
-        let pubkeys = (0..obf_params.input_size + 1)
-            .map(|idx| {
-                sample_public_key_by_idx(
+        let level_width_exp = obf_params.level_width_exp;
```

Comment:
@enricobottazzi 
Can you rename `level_width_exp` to `level_width`?
I think this "width" mentions how many inputs bits are inserted at each diamond.

Reply:
Correct, so I have two variable names :

- `level_width_exp` which refers to the exponent and indicates how many bits are inserted at each diamond step
- `level_width` which is equal to 2^level_width_exp and indicates the width in terms of the number of preimages to be generated at each step

How do you suggest to conciliate the two?

Reply:
How about renaming `level_width` to `level_size` or `num_level_preimages`?